packages/tinacms/src/admin/pages/CollectionCreatePage.tsx

```diff
@@ -37,7 +37,14 @@ const createDocument = async (
   const api = new TinaAdminApi(cms)
   const { filename, ...leftover } = values
 
-  const relativePath = `${folder ? `${folder}/` : ''}${filename}.${
+  if (typeof filename !== 'string') {
+    throw new Error('Filename must be a string')
+  }
+
+  // Append the folder if it exists and the filename does not start with a slash
+  const appendFolder = folder && !filename.startsWith('/') ? `/${folder}` : ''
+
+  const relativePath = `${appendFolder ? `${folder}/` : ''}${filename}.${
```

Comment:
```suggestion
  const relativePath = `${appendFolder}${filename}.${
```
I think you're already handing the empty case on the line above.

Reply:
Yeah! I think so. Ill push up this change.